File: Frontend/projectk-frontend/src/app/features/adminModule/models/userDto.ts
/**
 * The system-level roles, by name. The admin screen used to post the enum's ordinals
 * (Admin = 0, Member = 1), so reordering UserRole on the backend would have silently
 * assigned the wrong one.
 */
export type SystemUserRole = 'Admin' | 'Member';

export interface UserDto {
    userId: string;
    email: string;
    role: string;
    kurinKey: string | null;
    kurinNumber: number | null;
    firstName: string;
    lastName: string;
}

File: Backend/ProjectK.Backend/ProjectK.BusinessLogic/Modules/UsersModule/Models/UserDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectK.BusinessLogic.Modules.UsersModule.Models
{
    public class UserDto
    {
        public Guid UserId { get; set; }
        public Guid? KurinKey { get; set; }
        public int? KurinNumber { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string Role { get; set; }
    }
}
